src/Chapter3.hs

```diff
@@ -971,6 +1049,30 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday
+   = Mon
+   | Tue
+   | Wed
+   | Thu
+   | Fri
+   | Sat
+   | Sun
+   deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend w = case w of
+    Sat -> True
+    Sun -> True
+    _ -> False
+
+nextDay :: Weekday -> Weekday
+nextDay w
+    | w == maxBound = minBound
+    | otherwise = succ w
+
+daysToParty :: Weekday -> Int
+daysToParty w = abs (fromEnum w - fromEnum Fri)
```

Comment:
This would show the shortest distance to Friday, however, time doesn't work like this.
For example. if it is Saturday, it would tell you that it is 1 day to Friday, however, it is 6 days to go 🥳